src/services/apiKey.service.ts

```diff
@@ -0,0 +1,67 @@
+import { ApiKeyService as ApiKeyServiceAbstraction } from '../abstractions/apiKey.service';
+import { StorageService } from '../abstractions/storage.service';
+import { TokenService } from '../abstractions/token.service';
+
+import { Utils } from '../misc/utils';
+
+const Keys = {
+    clientId: 'clientId',
+    entityType: 'entityType',
+    entityId: 'entityId',
+};
+
+
+export class ApiKeyService implements ApiKeyServiceAbstraction {
+    private clientId: string;
+    private entityType: string;
+    private entityId: string;
+
+    constructor(private tokenService: TokenService, private storageService: StorageService) { }
+
+    setInformation(clientId: string) {
+        this.clientId = clientId;
+        const idParts = clientId.split('.');
+
+        if (idParts.length !== 2 || !Utils.isGuid(idParts[1])) {
+            throw Error('Invalid clientId');
+        }
+        this.entityType = idParts[0];
+        this.entityId = idParts[1];
+
+        return this.storageService.save(Keys.clientId, this.clientId)
```

Comment:
Why not just make 3 separate calls using `await`?

Reply:
I liked how the language makes it clear things are sequential, but I think that's just in juxtaposition to the `Promise.all` language.

normally I'd go this route so I'll change it